--- minetest/src/filesys.cpp	2025-05-12 15:16:01.198416862 +0200
+++ minetest/src/filesys.cpp	2025-05-12 14:26:59.447164472 +0200
@@ -28,6 +28,7 @@
 #include "log.h"
 #include "config.h"
 #include "porting.h"
+#include "notify_fs.h"
 #ifndef SERVER
 #include "irr_ptr.h"
 #endif
@@ -100,8 +101,10 @@
 bool CreateDir(const std::string &path)
 {
 	bool r = CreateDirectory(path.c_str(), NULL);
-	if(r == true)
+	if(r == true) {
+		notify_file_modified(path);
 		return true;
+	}
 	if(GetLastError() == ERROR_ALREADY_EXISTS)
 		return true;
 	return false;
@@ -145,6 +148,7 @@
 					<< path << std::endl;
 			return false;
 		}
+		notify_file_deleted(path);
 		return true;
 	}
 	infostream << "RecursiveDelete: Deleting content of directory "
@@ -164,6 +168,7 @@
 				<< path << std::endl;
 		return false;
 	}
+	notify_file_deleted(path);
 	return true;
 }
 
@@ -282,15 +287,15 @@
 bool CreateDir(const std::string &path)
 {
 	int r = mkdir(path.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
-	if (r == 0) {
+	if(r == 0) {
+		notify_file_modified(path);
 		return true;
 	}
-
+	
 	// If already exists, return true
 	if (errno == EEXIST)
 		return true;
 	return false;
-
 }
 
 bool PathExists(const std::string &path)
@@ -333,6 +338,7 @@
 					<< path << std::endl;
 			return false;
 		}
+		notify_file_deleted(path);
 		return true;
 	}
 	infostream << "RecursiveDelete: Deleting content of directory "
@@ -352,6 +358,7 @@
 				<< path << std::endl;
 		return false;
 	}
+	notify_file_deleted(path);
 	return true;
 }
 
@@ -851,6 +858,7 @@
 		return false;
 	}
 
+	notify_file_modified(path);
 	return true;
 }
 
@@ -932,7 +940,12 @@
 
 bool Rename(const std::string &from, const std::string &to)
 {
-	return rename(from.c_str(), to.c_str()) == 0;
+	bool success = rename(from.c_str(), to.c_str()) == 0;
+	if (success) {
+		notify_file_deleted(from);
+		notify_file_modified(to);
+	}
+	return success;
 }
 
 } // namespace fs
